src/arrayMethodJoin.js

```diff
@@ -4,8 +4,41 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+    let addToString = '';
+    let sep = separator;
+
+    if (this.length === 0) {
+      return result;
+    }
+
+    if (sep === null) {
+      sep = 'null';
+    }
+
+    for (let i = 0; i < this.length; i++) {
+      if (i === this.length - 1) {
+        if (this[i] === undefined) {
+          result = result + '';
+
+          return result;
+        }
+
+        result = result + this[i];
+
+        return result;
+      }
+
+      if (this[i] === null || this[i] === undefined) {
+        addToString = '' + sep;
+      } else {
+        addToString = this[i] + sep;
+      }
+      result = result + addToString;
+    }
+
+    return result;
```

Comment:
you have so much conditions, but actually you need only two.
if (this[i] !== null || this[i] !== undefined)
add this[i] to the resulting string, than if your element is not last - add separator to the resulting string, that's all)